crates/hyperdrive-math/src/long/targeted.rs

```diff
@@ -0,0 +1,503 @@
+
+        // a'(x) = u / c - g'(x)
+        let inner_numerator_derivative = self.mu() / self.vault_share_price() - gov_fee_derivative;
+
+        // b(x) = y_0 - Y(x)
+        let inner_denominator = self.bond_reserves() - bond_amount;
+
+        // b'(x) = Y'(x)
+        let long_amount_derivative = match self.long_amount_derivative(base_amount) {
+            Some(derivative) => derivative,
+            None => return Err(eyre!("long_amount_derivative failure.")),
+        };
+
+        // v(x) = a(x) / b(x)
+        // v'(x) = ( b(x) * a'(x) + a(x) * b'(x) ) / b(x)^2
+        let inner_derivative = (inner_denominator * inner_numerator_derivative
+            + inner_numerator * long_amount_derivative)
+            / inner_denominator.pow(fixed!(2e18));
+
+        // p'(x) = v'(x) T v(x)^(T-1)
+        // p'(x) = v'(x) T v(x)^(-1)^(1-T)
+        // v(x) is flipped to (denominator / numerator) to avoid a negative exponent
+        Ok(inner_derivative
+            * self.time_stretch()
+            * (inner_denominator / inner_numerator).pow(fixed!(1e18) - self.time_stretch()))
+    }
+
+    /// Calculate the base & bond deltas from the current state given desired new reserve levels.
+    ///
+    /// Given a target ending pool share reserves, $z_t$, and bond reserves, $y_t$,
+    /// the trade deltas to achieve that state would be:
+    ///
+    /// $$
+    /// \Delta x = c * (z_t - z_e) \\
+    /// \Delta y = y - y_t - c(\Delta x)
+    /// $$
+    ///
+    /// where $c$ is the vault share price and
+    /// $c(\Delta x)$ is the (open_long_curve_fee)[long::fees::open_long_curve_fees].
+    fn trade_deltas_from_reserves(
+        &self,
+        share_reserves: FixedPoint,
+        bond_reserves: FixedPoint,
+    ) -> (FixedPoint, FixedPoint) {
+        let base_delta =
+            (share_reserves - self.effective_share_reserves()) * self.vault_share_price();
+        let bond_delta =
+            (self.bond_reserves() - bond_reserves) - self.open_long_curve_fees(base_delta);
+        (base_delta, bond_delta)
+    }
+
+    /// Calculates the pool reserve levels to achieve a target interest rate.
+    /// This calculation does not take Hyperdrive's solvency constraints or exposure
+    /// into account and shouldn't be used directly.
+    ///
+    /// The price for a given fixed-rate is given by $p = 1 / (r t + 1)$, where
+    /// $r$ is the fixed-rate and $t$ is the annualized position duration. The
+    /// price for a given pool reserves is given by $p = \frac{\mu z}{y}^T$,
+    /// where $\mu$ is the initial share price and $T$ is the time stretch
+    /// constant. By setting these equal we can solve for the pool reserve levels
+    /// as a function of a target rate.
+    ///
+    /// For some target rate, $r_t$, the pool share reserves, $z_t$, must be:
+    ///
+    /// $$
+    /// z_t = \frac{1}{\mu} \left(
+    ///   \frac{k}{\frac{c}{\mu} + \left(
+    ///     (r_t t + 1)^{\frac{1}{T}}
+    ///   \right)^{1 - T}}
```

Comment:
IMO, it would be clearer to simplify the exponents to $\frac{1 - T}{T}$ (or $\frac{1 - t_s}{t_s}$ if we switch to the usual notation).

Reply:
This also minimizes the invocations of `pow`, which reduces error and makes the calculation more efficient.

Reply:
I definitely could be missing something, but I think the number of `pow` invocations stays the same because the `target_bond_reserves` do not want that simplification.

Here's a modified version using the simplified exponent, which still has to have 3 `pow` calls.

```rs
        let c_over_mu = self
            .vault_share_price()
            .div_up(self.initial_vault_share_price());
        let scaled_rate = target_rate.mul_up(self.annualized_position_duration()) + fixed!(1e18);
        let inner = (self.k_down()
            / (c_over_mu
                + scaled_rate.pow((fixed!(1e18) - self.time_stretch()) / self.time_stretch())))
        .pow(fixed!(1e18) / (fixed!(1e18) - self.time_stretch()));
        let target_share_reserves = inner / self.initial_vault_share_price();

        // Then get the target bond reserves.
        let target_bond_reserves = inner * scaled_rate.pow(fixed!(1e18) / self.time_stretch());
```

You can see that here:

$$
z_{t} = \tfrac{1}{\mu} \left( \tfrac{k}{ \tfrac{c}{\mu} + \left( (r_{t} t_{d}+1)^{\tfrac{1}{t_{s}}} \right)^{1-t_{s}}} \right)^{\tfrac{1}{1-t_{s}}}
$$

$$
y_{t} = \left( \tfrac{k}{ \tfrac{c}{\mu} +  \left( \left( r_{t} t_{d} + 1 \right)^{\tfrac{1}{t_{s}}} \right)^{1-t_{s}}} \right)^{1-t_{s}} \left( r_{t} t_{d} + 1 \right)^{\tfrac{1}{t_{s}}}
$$

The $y_{t}$ term wants the exponent to be $\tfrac{1}{t_s}$ and the $z_{t}$ term wants the exponent to be $\frac{1 - t_s}{t_s}$. By keeping them separated, I can simplify the `target_bond_reserves` equation.

I think the difference is minor, and I'm happy to go with your suggestion if you still want it.  Unless I missed a way to reduce the `pow`s; if so please let me know.

Reply:
I'm talking about the inner exponents. This term:

$$
\left( \left( 1 + r_t \cdot t_d \right)^{\tfrac{1}{t_s}} \right)^{1 - t_s}
$$

We can combine the two exponents into $\tfrac{1 - t_s}{t_s}$ in both.

Reply:
resolved in a phone call; same number of pows so I'll leave it as is.